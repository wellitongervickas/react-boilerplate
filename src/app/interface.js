// @flow strict

export interface Props {
  setRootInitialized: Function;
  setRootSagasInitialized: Function;
  rootInitialized: boolean;
  rootSagasInitialized: boolean;
}
