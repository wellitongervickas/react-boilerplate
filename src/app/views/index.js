import Welcome from './welcome';
import NotFound from './not-found';

export {
  Welcome,
  NotFound,
};
