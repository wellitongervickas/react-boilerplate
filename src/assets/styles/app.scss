// Bootstrap Styles
@import '~bootstrap/scss/_functions.scss';
@import '~bootstrap/scss/_variables.scss';
@import '~bootstrap/scss/_mixins.scss';
@import '~bootstrap/scss/_root.scss';
@import '~bootstrap/scss/_reboot.scss';
@import '~bootstrap/scss/_type.scss';
@import '~bootstrap/scss/_utilities.scss';
@import '~bootstrap/scss/_grid.scss';


// App Styles
@import './base/base';
@import './layout/layout';
@import './utilities/utilities';
