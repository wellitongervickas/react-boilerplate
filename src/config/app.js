import ptBr from '../locales/pt-BR.json';

export default {
  appName: 'React Boilerplate',
  poweredBy: 'Welliton Gervickas',
  lang: ptBr,
};
