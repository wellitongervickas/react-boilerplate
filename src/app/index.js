// @flow strict

import * as React from 'react';
import type { Node } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Route, Switch } from 'react-router';
import { ConnectedRouter } from 'connected-react-router';
import * as rootActions from '../store/actions/root';
import config from '../config';
import { history } from '../store';
import { Props } from './interface';
import { RoutesInterface } from '../utils/interfaces/routes/routes-interface';

export class App extends React.Component<Props> {
  static defaultProps: Props = {
    setRootInitialized: () => { },
    setRootSagasInitialized: () => { },
    rootInitialized: false,
    rootSagasInitialized: false,
  }

  componentWillMount = () => {
    // Test to initialize store in react debug mode
    const { setRootInitialized, setRootSagasInitialized } = this.props;
    setRootInitialized(true);
    setRootSagasInitialized(true);
  }

  renderMain = (): Node => [
    <section key="header">
      <header>Header</header>
    </section>,
    <section key="content">
      <main>
        <Switch history={history}>
          {config.routes.map<RoutesInterface>(item => <Route key={item.key} {...item} />)}
        </Switch>
      </main>
    </section>,
    <section key="footer">
      <footer>Footer</footer>
    </section>,
  ]

  render = (): Node => (
    <ConnectedRouter history={history}>
      <div>
        {this.renderMain()}
      </div>
    </ConnectedRouter>
  )
}

const mapStateToProps = state => ({
  rootInitialized: state.root.rootInitialized,
  rootSagasInitialized: state.root.rootSagasInitialized,
});

const mapDispatchToProps = dispatch => bindActionCreators(rootActions, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(App);
