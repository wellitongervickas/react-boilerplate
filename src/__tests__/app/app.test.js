import React from 'react';
import { shallow } from 'enzyme';
import { App } from '../../app';
import renderer from 'react-test-renderer';
import { mapStateToProps, mapDispatchToProps } from '../../app';

describe('App', () => {

  const initialState = {
    root: {
      rootInitialized: false,
      rootSagasInitialized: false,

    }
  }

  it('renders without crashing', () => {
    shallow(<App />);
  });

  it('renders with snapshot', () => {
    const tree = shallow(<App />).dive()
    expect(tree).toMatchSnapshot();
  });

  it('should test initial map state to props', () => {
    expect(mapStateToProps(initialState).rootInitialized).toEqual(false);
    expect(mapStateToProps(initialState).rootSagasInitialized).toEqual(false);
  })

  it('should test initial map actions to props', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).setRootInitialized();
    mapDispatchToProps(dispatch).setRootSagasInitialized();
    expect(dispatch.mock.calls.length).toBe(2)
  })
})