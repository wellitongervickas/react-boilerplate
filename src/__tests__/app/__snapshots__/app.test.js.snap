// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(ConnectedRouterWithContext)
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <div>
      <section>
        <header>
          Header
        </header>
      </section>
      <section>
        <main>
          <Switch
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
          >
            <Route
              component={[Function]}
              exact={true}
              name="WELCOME"
              path="/"
            />
            <Route
              component={[Function]}
              name="NOT_FOUND_PAGE"
              path="/not-found"
            />
            <Route
              component={[Function]}
            />
          </Switch>
        </main>
      </section>
      <section>
        <footer>
          Footer
        </footer>
      </section>
    </div>
  </Connect(ConnectedRouterWithContext)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {},
  },
  Symbol(enzyme.__childContext__): null,
}
`;
